### ISSUE TITLE
solveset(Abs(x) - n) should return {-n, n}, only when n is positive

### ISSUE THREAD

**Issue by aktech** [sympy/sympy#9588]
```python
In [8]: x = Symbol('x', real=True)

In [9]: n = Symbol('n', real=True)

In [12]: solveset(Abs(x) - n, x)
Out[12]: {-n, n}
```

For general `n` it must tell somehow that the solution is for positive `n`.

**Comment by aktech** [comment]
Also:

```python
In [39]: k = Abs(x**2 - 1) - y

In [40]: solveset(k, x)
Out[40]: 
    ⎧   ________    ________     _______    _______ ⎫
ℝ ∩ ⎨-╲╱ -y + 1 , ╲╱ -y + 1 , -╲╱ y + 1 , ╲╱ y + 1 ⎬
    ⎩                                               ⎭

```

**Comment by hargup** [comment]
The main issue was fixed with #9628 but now we have:
```python
In [10]: k = Abs(x**2 - 1) - y

In [11]: solveset(k, x)
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-11-e766d2e03b92> in <module>()
----> 1 solveset(k, x)

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in solveset(f, symbol)
    894     if isinstance(f, (Expr, Number)):
    895         if real is True:
--> 896             return solveset_real(f, symbol)
    897         else:
    898             return solveset_complex(f, symbol)

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in solveset_real(f, symbol)
    468         return FiniteSet(*result).intersect(S.Reals)
    469     else:
--> 470         return result.intersect(S.Reals)
    471 
    472 

/home/hargup/210fs/sympy/sympy/sets/sets.py in intersect(self, other)
    101 
    102         """
--> 103         return Intersection(self, other)
    104 
    105     def intersection(self, other):

/home/hargup/210fs/sympy/sympy/sets/sets.py in __new__(cls, *args, **kwargs)
   1381         # Reduce sets using known rules
   1382         if evaluate:
-> 1383             return Intersection.reduce(args)
   1384 
   1385         return Basic.__new__(cls, *args)

/home/hargup/210fs/sympy/sympy/sets/sets.py in reduce(args)
   1452                 if len(other_sets) > 0:
   1453                     other = Intersection(other_sets)
-> 1454                     return Union(Intersection(arg, other) for arg in s.args)
   1455                 else:
   1456                     return Union(arg for arg in s.args)

/home/hargup/210fs/sympy/sympy/sets/sets.py in __new__(cls, *args, **kwargs)
   1145                 return sum(map(flatten, arg), [])
   1146             raise TypeError("Input must be Sets or iterables of Sets")
-> 1147         args = flatten(args)
   1148 
   1149         # Union of no sets is EmptySet

/home/hargup/210fs/sympy/sympy/sets/sets.py in flatten(arg)
   1143                     return [arg]
   1144             if iterable(arg):  # and not isinstance(arg, Set) (implicit)
-> 1145                 return sum(map(flatten, arg), [])
   1146             raise TypeError("Input must be Sets or iterables of Sets")
   1147         args = flatten(args)

/home/hargup/210fs/sympy/sympy/sets/sets.py in flatten(arg)
   1143                     return [arg]
   1144             if iterable(arg):  # and not isinstance(arg, Set) (implicit)
-> 1145                 return sum(map(flatten, arg), [])
   1146             raise TypeError("Input must be Sets or iterables of Sets")
   1147         args = flatten(args)

/home/hargup/210fs/sympy/sympy/sets/sets.py in <genexpr>(.0)
   1452                 if len(other_sets) > 0:
   1453                     other = Intersection(other_sets)
-> 1454                     return Union(Intersection(arg, other) for arg in s.args)
   1455                 else:
   1456                     return Union(arg for arg in s.args)

/home/hargup/210fs/sympy/sympy/sets/sets.py in __new__(cls, *args, **kwargs)
   1381         # Reduce sets using known rules
   1382         if evaluate:
-> 1383             return Intersection.reduce(args)
   1384 
   1385         return Basic.__new__(cls, *args)

/home/hargup/210fs/sympy/sympy/sets/sets.py in reduce(args)
   1440                        if any(other.contains(x) not in (True, False) for other in other_args)]
   1441                 if unk:
-> 1442                     other_sets = Intersection(*other_args)
   1443                     if other_sets.is_EmptySet:
   1444                         return EmptySet()

/home/hargup/210fs/sympy/sympy/sets/sets.py in __new__(cls, *args, **kwargs)
   1381         # Reduce sets using known rules
   1382         if evaluate:
-> 1383             return Intersection.reduce(args)
   1384 
   1385         return Basic.__new__(cls, *args)

/home/hargup/210fs/sympy/sympy/sets/sets.py in reduce(args)
   1473                 new_args = False
   1474                 for t in args - set((s,)):
-> 1475                     new_set = s._intersect(t)
   1476                     # This returns None if s does not know how to intersect
   1477                     # with t. Returns the newly intersected set otherwise

/home/hargup/210fs/sympy/sympy/sets/fancysets.py in _intersect(self, other)
    299             return imageset(Lambda(n_, re),
    300                             self.base_set.intersect(
--> 301                                 solveset_real(im, n_)))
    302 
    303 

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in solveset_real(f, symbol)
    450                         result += _solve_radical(equation,
    451                                                  symbol,
--> 452                                                  solveset_real)
    453                     elif equation.has(Abs):
    454                         result += _solve_abs(f, symbol)

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in _solve_radical(f, symbol, solveset_solver)
    632     eq, cov = unrad(f)
    633     if not cov:
--> 634         result = solveset_solver(eq, symbol) - \
    635             Union(*[solveset_solver(g, symbol) for g in denoms(f, [symbol])])
    636     else:

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in solveset_real(f, symbol)
    452                                                  solveset_real)
    453                     elif equation.has(Abs):
--> 454                         result += _solve_abs(f, symbol)
    455                     else:
    456                         result += _solve_as_rational(equation, symbol,

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in _solve_abs(f, symbol)
    663 
    664         sols_q_pos = solveset_real(f_p*f_q + f_r,
--> 665                                            symbol).intersect(q_pos_cond)
    666         sols_q_neg = solveset_real(f_p*(-f_q) + f_r,
    667                                            symbol).intersect(q_neg_cond)

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in solveset_real(f, symbol)
    428         # wrong solutions we are using this technique only if both f and g are
    429         # finite for a finite input.
--> 430         result = Union(*[solveset_real(m, symbol) for m in f.args])
    431     elif _is_function_class_equation(TrigonometricFunction, f, symbol) or \
    432             _is_function_class_equation(HyperbolicFunction, f, symbol):

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in <listcomp>(.0)
    428         # wrong solutions we are using this technique only if both f and g are
    429         # finite for a finite input.
--> 430         result = Union(*[solveset_real(m, symbol) for m in f.args])
    431     elif _is_function_class_equation(TrigonometricFunction, f, symbol) or \
    432             _is_function_class_equation(HyperbolicFunction, f, symbol):

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in solveset_real(f, symbol)
    456                         result += _solve_as_rational(equation, symbol,
    457                                                      solveset_solver=solveset_real,
--> 458                                                      as_poly_solver=_solve_as_poly_real)
    459                 else:
    460                     result += solveset_real(equation, symbol)

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in _solve_as_rational(f, symbol, solveset_solver, as_poly_solver)
    476     g, h = fraction(f)
    477     if not h.has(symbol):
--> 478         return as_poly_solver(g, symbol)
    479     else:
    480         valid_solns = solveset_solver(g, symbol)

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in _solve_as_poly_real(f, symbol)
    579     return _solve_as_poly(f, symbol,
    580                           solveset_solver=solveset_real,
--> 581                           invert_func=invert_real)
    582 
    583 

/home/hargup/210fs/sympy/sympy/solvers/solveset.py in _solve_as_poly(f, symbol, solveset_solver, invert_func)
    552                 else:
    553                     raise NotImplementedError(
--> 554                         "inversion of %s not handled" % gen)
    555         else:
    556             raise NotImplementedError("multiple generators not handled"

NotImplementedError: inversion of sin(atan2(0, _n)/2) not handled
```
